### ISSUE TITLE
OSError: [Errno 99] Cannot assign requested address

### ISSUE THREAD

**Issue by mathieuchateau** [ddbnl/office365-audit-log-collector#6]
Hello,

During initial collection, I had this crash.
Tenant is big, like 11K users.

I guess I can start it again, but I don't know if it means a gap in log collection, related to known_content file ?

```
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/office365-audit-log-collector-master/GraylogInterface.py", line 36, in monitor_queue
    self._send_message_to_graylog(msg=msg)
  File "/home/ubuntu/office365-audit-log-collector-master/GraylogInterface.py", line 60, in _send_message_to_graylog
    sock = self._connect_to_graylog_input()
  File "/home/ubuntu/office365-audit-log-collector-master/GraylogInterface.py", line 48, in _connect_to_graylog_input
    s.connect((self.gl_address, int(self.gl_port)))
OSError: [Errno 99] Cannot assign requested address
```

**Comment by mathieuchateau** [comment]
I could work around this issue by tweaking linux kernel.
When it fails, and is started again, I found duplicated exact messages, so I think it's important to workaround it. Maybe implement a 30s sleep and 1 retry when it occurs ?

After tweaking tcp settings, I . had around 23K TCP connections from python script to Graylog TCP input. By default, TCP wait time is 60s and TCP connection are not reused, which lead to tcp connections overload.

Here are settings I added to /etc/sysctl.conf to handle load without error. I am using GrayLog OVA.
the last two settings are the most important ones to avoid this error I think.
Then do a sysctl -p, no need to reboot.

As a side note, to get all data and not only the most recent one, I edited AuditLogCollector.py to change hours=1 to days=2. I don't know why doing with more than 2 generates errors, but using this, I could gather 1 month of logs.

````
net.core.rmem_max = 16777216
net.core.wmem_max = 16777216
net.ipv4.tcp_rmem = 4096 87380 16777216
net.ipv4.tcp_wmem = 4096 65536 16777216
net.ipv4.tcp_syncookies = 1
net.ipv4.tcp_mem = 50576   64768   98152
net.core.netdev_max_backlog = 2500
net.ipv4.tcp_max_syn_backlog = 100000
net.core.somaxconn = 100000
net.core.netdev_max_backlog = 100000
net.ipv4.ip_local_port_range = 18000    65535
net.ipv4.tcp_fin_timeout = 5
net.ipv4.tcp_tw_reuse = 1
```
